test/dns/dns_api_test.rb

```diff
@@ -32,53 +39,173 @@ def app
   end
 
   def test_create_a_record
-    post '/', :fqdn => 'test.com', :value => '192.168.33.33', :type => 'A'
-    assert_equal 'test.com', @server.fqdn
-    assert_equal '192.168.33.33', @server.ip
+    fqdn = get_fqdn
+    ip = get_ip
+    post '/', :fqdn => fqdn, :value => ip, :type => 'A'
+    assert_equal fqdn, @server.fqdn
+    assert_equal ip, @server.ip
     assert_equal 'A', @server.type
   end
 
   def test_create_returns_error_if_fqdn_is_missing
-    post '/', :value => '192.168.33.33', :type => 'A'
+    post '/', :value => get_ip, :type => 'A'
     assert_equal 400, last_response.status
   end
 
   def test_create_returns_error_if_ip_is_missing
-    post '/', :fqdn => 'test.com', :type => 'A'
+    post '/', :fqdn => get_fqdn, :type => 'A'
     assert_equal 400, last_response.status
   end
 
   def test_create_returns_error_if_type_is_missing
-    post '/', :fqdn => 'test.com', :value => '192.168.33.33'
+    post '/', :fqdn => get_fqdn, :value => get_ip
     assert_equal 400, last_response.status
   end
 
   def test_create_returns_error_if_type_is_unrecognized
-    post '/', :fqdn => 'test.com', :value => '192.168.33.33', :type => "FFF"
+    post '/', :fqdn => get_fqdn, :value => get_ip, :type => "FFF"
+    assert_equal 400, last_response.status
+  end
+
+  def test_create_returns_error_if_invalid_name_for_a
```

Comment:
Could you create unit-type tests for the name validator, and keep one integration-level test for a valid fqdn and one for an invalid one instead?